javascript-bank/bank.js

```diff
@@ -0,0 +1,40 @@
+/* exported Bank */
+function Bank() {
+  this.nextAccountNumber = 1;
+  this.accounts = [];
+}
+
+Bank.prototype.openAccount = function (holder, balance) {
+  if (balance > 0) {
+    var account = new Account(this.nextAccountNumber, holder);
+    account.deposit(balance);
+    this.accounts.push(account);
+    this.nextAccountNumber++;
+    return account.number;
+  } else {
+    return null;
+  }
+};
+
+Bank.prototype.getAccount = function (number) {
+  for (var i = 0; i < this.accounts.length; i++) {
+    if (this.accounts[i].number === number) {
+      return this.accounts[i];
+    }
+  }
+  if (number !== this.accounts.number) {
```

Comment:
`this.accounts` is an array, which does not have a `number` property. You can just `return null` here without the `if` statement, since, the loop would have found it by now.